### PR TITLE
Scala 2.13.0-M3

### PR DESCRIPTION
**Pull request by xuwei-k**
# Pull Request Checklist

* [x] I agree to licence my contributions under the [MIT licence](https://github.com/paulbutcher/ScalaMock/blob/master/LICENCE)
* [x] I have added copyright headers to new files
* [x] I have added tests for any changed functionality

## Purpose

- Update build to include 2.13.0-M3
- Update version of specs2 (except for Scala 2.10 build)

### CODE REVIEW

File: examples/shared/src/test/scala/com/example/makro/ControllerTest.scala [hunk 1]
```diff
@@ -33,18 +33,18 @@ class ControllerTest extends FunSuite with MockFactory {
  
     inSequence {
       inAnyOrder {
-        (mockTurtle.penUp _) expects ()
+        (() => mockTurtle.penUp) expects ()
```
Comment (barkhorn) [inline examples/shared/src/test/scala/com/example/makro/ControllerTest.scala]: why has this style changed? Is this required by Scala 2.13?

File: build.sbt [hunk 2]
```diff
@@ -12,8 +12,16 @@ developers in ThisBuild := List(
 )
 homepage in ThisBuild := Some(url("http://scalamock.org/"))
 
-lazy val scalatest = "org.scalatest" %% "scalatest" % "3.0.3"
-lazy val specs2 = "org.specs2" %% "specs2-core" % "3.9.1"
+lazy val scalatest = "org.scalatest" %% "scalatest" % "3.0.5-M1"
+lazy val specs2 = Def.setting {
+  val v = CrossVersion.partialVersion(scalaVersion.value) match {
+    case Some((2, 10)) =>
+      "3.9.1" // specs2 4.x does not support Scala 2.10
```
Comment (barkhorn) [inline build.sbt]: Are 3.9.x and 4.0.x guaranteed to be API compatible?

**Comment by barkhorn** [comment]
Hi, I've added the MR template back.
Could you have a look at my comments please, and acknowledge the checklist for the MR if you are happy with it?

File: examples/shared/src/test/scala/com/example/makro/ControllerTest.scala (hunk 1 above)
Comment (xuwei-k) [reply examples/shared/src/test/scala/com/example/makro/ControllerTest.scala]: required by Scala 2.13. old style is compile error

File: build.sbt (hunk 2 above)
Comment (xuwei-k) [reply build.sbt]: I think so

**Pull request merged**